File: C1.UWP.FlexGrid/VB/FlexGridSamples/CellFactory/FinancialCellFactory.vb
Imports Windows.UI.Xaml.Data
Imports Windows.UI.Xaml.Controls
Imports Windows.UI.Xaml
Imports C1.Xaml.FlexGrid
Imports System.Reflection

Public Class FinancialCellFactory
    Inherits CellFactory
    Shared _thicknessEmpty As New Thickness(0)

    ' bind cell to ticker
    Public Overrides Sub CreateCellContent(grid As C1FlexGrid, bdr As Border, range As CellRange)
        ' create visual element for this cell
        Dim dataItem As Object = grid.Rows(range.Row).DataItem
        Dim name As String = grid.Columns(range.Column).PropertyInfo.Name

        If TypeOf dataItem Is FinancialData AndAlso (name.Equals("LastSale") OrElse name.Equals("Bid") OrElse name.Equals("Ask")) Then
            ' create stock ticker cell
            Dim ticker As New StockTicker()
            bdr.Child = ticker
            bdr.Padding = _thicknessEmpty
        Else
            ' use default implementation
            MyBase.CreateCellContent(grid, bdr, range)
        End If
    End Sub

    Public Overrides Sub DisposeCell(grid As C1FlexGrid, cellType As CellType, cell As FrameworkElement)
        MyBase.DisposeCell(grid, cellType, cell)
    End Sub

    Public Sub ShowLiveData(grid As C1FlexGrid, range As CellRange, cell As FrameworkElement)
        'Sets the binding in the cell content so that is updated at runtime.
        Dim stockTicker As StockTicker = TryCast((TryCast(cell, Border)).Child, StockTicker)
        If stockTicker IsNot Nothing Then
            Dim c As Column = grid.Columns(range.Column)
            Dim r As Row = grid.Rows(range.Row)
            Dim pi As PropertyInfo = c.PropertyInfo

            ' to show sparklines
            stockTicker.Tag = r.DataItem
            stockTicker.BindingSource = pi.Name

            Dim binding As New Binding() With {
                .Path = New PropertyPath(pi.Name)
            }
            binding.Converter = New MyConverter()
            binding.Source = r.DataItem
            binding.Mode = BindingMode.OneWay
            stockTicker.SetBinding(stockTicker.ValueProperty, binding)
        End If
    End Sub
End Class

Public Class MyConverter
    Implements IValueConverter
    Public Function Convert(value As Object, targetType As System.Type, parameter As Object, language As String) As Object Implements IValueConverter.Convert
        Return System.Convert.ToDouble(value)
    End Function

    Public Function ConvertBack(value As Object, targetType As System.Type, parameter As Object, language As String) As Object Implements IValueConverter.ConvertBack
        Return value
    End Function
End Class

File: C1.UWP.FlexGrid/CS/FlexGrid101/CellFactory/FinancialCellFactory.cs
using C1.Xaml.FlexGrid;
using Windows.UI.Xaml;
using Windows.UI.Xaml.Controls;
using Windows.UI.Xaml.Data;

namespace FlexGrid101
{
    public class FinancialCellFactory : CellFactory
    {
        static Thickness _thicknessEmpty = new Thickness(0);

        // bind cell to ticker
        public override void CreateCellContent(C1FlexGrid grid, Border bdr, CellRange range)
        {
            // create visual element for this cell
            var dataItem = grid.Rows[range.Row].DataItem;
            var name = grid.Columns[range.Column].PropertyInfo.Name;
            if (dataItem is FinancialData &&
               (name.Equals("LastSale") || name.Equals("Bid") || name.Equals("Ask")))
            {
                // create stock ticker cell
                StockTicker ticker = new StockTicker();
                bdr.Child = ticker;
                bdr.Padding = _thicknessEmpty;
            }
            else
            {
                // use default implementation
                base.CreateCellContent(grid, bdr, range);
            }
        }

        public override void DisposeCell(C1FlexGrid grid, CellType cellType, FrameworkElement cell)
        {
            base.DisposeCell(grid, cellType, cell);
        }

        public void ShowLiveData(C1FlexGrid grid, CellRange range, FrameworkElement cell)
        {
            //Sets the binding in the cell content so that is updated at runtime.
            var stockTicker = (cell as Border).Child as StockTicker;
            if (stockTicker != null)
            {
                var c = grid.Columns[range.Column];
                var r = grid.Rows[range.Row];
                var pi = c.PropertyInfo;

                // to show sparklines
                stockTicker.Tag = r.DataItem;
                stockTicker.BindingSource = pi.Name;

                var binding = new Binding { Path = new PropertyPath(pi.Name) };
                binding.Converter = new MyConverter();
                binding.Source = r.DataItem;
                binding.Mode = BindingMode.OneWay;
                stockTicker.SetBinding(StockTicker.ValueProperty, binding);
            }
        }
    }

    public class MyConverter : IValueConverter
    {
        public object Convert(object value, System.Type targetType, object parameter, string language)
        {
            return System.Convert.ToDouble(value);
        }

        public object ConvertBack(object value, System.Type targetType, object parameter, string language)
        {
            return value;
        }
    }
}
